server/chat-plugins/battlesearch.ts

```diff
@@ -0,0 +1,412 @@
+/**
+ * Battle search - handles searching battle logs.
+ */
+import * as child_process from 'child_process';
+import * as util from 'util';
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {Dex} from '../../sim/dex';
+import {Config} from '../config-loader';
```

Comment:
Yeesh, I didn't know this was done this way. I wasn't aware Config was designed to be imported this way...

Reply:
I think we should import the old way for `Dex` and `Config`. That's more consistent to the rest of the codebase and allows them to be hotpatched with their corresponding command.

Reply:
What do you mean, "the old way"?

Reply:
Like we do in `room-battle` etc:

https://github.com/smogon/pokemon-showdown/blob/52fb644d9eb57acbeaaaa2b60cf670cec21c6e5f/server/room-battle.ts#L1267-L1271